### PR TITLE
misc. incl. high ang. mom. basis sets

### PR DESCRIPTION
**Pull request by loriab**
## Description
misc.

## Todos
- [x] addressed discrepancy in high ang mom conventions in basis sets. `L=7` and so forth is favored, but HIJ and HIK both accepted. internal basis set library transformed: 'L=7' <- 'K' . see https://github.com/MolSSI-BSE/basis_set_exchange/issues/121, see https://github.com/psi4/psi4/issues/628, closes https://github.com/psi4/psi4/issues/1992 @susilehtola @bennybp 
- [x] consolidated the weird-basis tests (skip AM) into pytest
- [x] @CDSherrill and @edeprince3 confirmed that one should use the low mem algorithm in fnocc for performance, so changed the default
- [x] lets one run pcmsolver through schema by replacing pcm_helper closes https://github.com/psi4/psi4/issues/1986 . works, but hit problem where you can't start pcmsolver the py module afresh, so it complains about units already set. @robertodr contacted
- [x] libiomp5 (Intel) and libomp (LLVM) are interchangeable. relaxed a test on conda packages for mac to accommodate.

## Questions
- [ ] anyone else having problems with mbis3 quadrupole?

## Checklist
- [x] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Review by andysim — APPROVED** [review]
Yay negative LoC!  That pytest cleanup really simplifies things.  LGTM

**Comment by loriab** [comment]
yup, I love `pytest.mark.parametrize`. I don't really have a plan for `output.ref` in pytest-land, hence the negative LoC.

**Comment by hokru** [comment]
Surprising that the low memory algorithm is faster! 

I think in that case the output should be adapted to avoid confusion as there are warnings printed. Maybe also 'regular algorithm' can have a different name, but it's what other programs do, I suppose.

https://github.com/psi4/psi4/blob/master/psi4/src/psi4/fnocc/lowmemory_triples.cc#L72
https://github.com/psi4/psi4/blob/master/psi4/src/psi4/fnocc/df_ccsd.cc#L659-L661
https://github.com/psi4/psi4/blob/master/psi4/src/psi4/fnocc/ccsd.cc#L780

File: doc/sphinxman/source/basissets.rst [hunk 1]
```diff
@@ -83,9 +83,13 @@ Appendix :ref:`apdx:basisElement`.
 
 |PSIfour| uses the angular momentum convention below that, consistent
 with EMSL, skips the letter ``J``. Note that Gaussian94 convention is
-*not* to skip this letter. Another portion of the G94 format, labeling
-angular momentum with :samp:`L={l}` syntax is not presently implemented,
-though this is coming. ::
+*not* to skip this letter.
```
Comment (susilehtola) [inline doc/sphinxman/source/basissets.rst:86]: Remove mention of EMSL; BSE does not skip the letter

File: doc/sphinxman/source/basissets.rst [hunk 2]
```diff
@@ -83,9 +83,13 @@ Appendix :ref:`apdx:basisElement`.
...
+|PSIfour| can use either convention for ``.gbs`` import (assuming
+angular momentum levels are not skipped), but it will always output
+in the ``HIK`` sequence. Another portion of the G94 format, labeling
+angular momentum with :samp:`L={l}` syntax is newly (August 2020)
+presently implemented, and this is preferred for ``L=7`` and above to
```
Comment (susilehtola) [inline doc/sphinxman/source/basissets.rst:91]: "newly presently"?

File: psi4/driver/qcdb/libmintsbasissetparser.py [hunk 3]
```diff
@@ -246,7 +255,20 @@ def parse(self, symbol, dataset):
...
+                                            am = shell_to_am_HIK[shell_type[0]]
+                                        if am_convention == "HIJ" or (am == -1 and shell_type[0].upper() == "J" and max_am_to_date == 6):
+                                            am_convention = "HIJ"
+                                            am = shell_to_am_HIJ[shell_type[0]]
+                                        if am_convention is None and am == -1:
+                                            raise ValidationError(f"""Gaussian94BasisSetParser::parse: angular momentum type {shell_type[0]} too high and HIJ/HIK convention uncertain. Use, for example, `L=7` notation. line {lineno}: {line}""")
+                                    else:
+                                        am = int(shell_type[2:])
+
+                                    max_am_to_date = max(max_am_to_date, am)
```
Comment (susilehtola) [inline psi4/driver/qcdb/libmintsbasissetparser.py:271]: It's perfectly legal to have the shells in arbitrary order, e.g. P, S, D, G, K, I, F...

Comment (loriab) [reply psi4/driver/qcdb/libmintsbasissetparser.py:271]: right. I added a few tests to show this still works (plus your weird basis ones). Only time when it won't work is if the HIK/HIJ convention hasn't been hinted beforehand. As that's the intersection of AM>7 letter-based basis set and weirdly ordered basis set, I'm not too worried.

**Comment by loriab** [comment]
> I think in that case the output should be adapted to avoid confusion as there are warnings printed. Maybe also 'regular algorithm' can have a different name, but it's what other programs do, I suppose.

Yes, there's some graphs to prove it I should dig up. Thanks for reading the printouts -- I glanced at the logic but not the warnings. Now memory and algorithm are printed with "high-memory"/"low-memory" labels and without efficiency judgements.

**Comment by loriab** [comment]
## high-mem vs low-mem algorithm for (T) in FNOCC

[bz2-fnocc-t-time.pdf](https://github.com/psi4/psi4/files/5178403/bz2-fnocc-t-time.pdf)
[bz2-fnocc-t-effic.pdf](https://github.com/psi4/psi4/files/5178404/bz2-fnocc-t-effic.pdf)

**Comment by hokru** [comment]
`mbis-3` fails also on my mac.

**Review by hokru — APPROVED** [review]
lgtm. The basis set parser is not getting simpler.

File: tests/pytests/test_weird_basis.py [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+ 1.0367000000e+00  1.0000000000e+00
+{L9} 1 1.0
+ 4.7209680000e+00  1.0000000000e+00
+{L9} 1 1.0
+ 1.3433090000e+00  1.0000000000e+00
+****
+"""
+
+
+_hijhik = pytest.mark.xfail(reason="HIJ/HIK convention uncertail", raises=psi4.driver.qcdb.exceptions.ValidationError, strict=True)
```
Comment (PeterKraus) [inline tests/pytests/test_weird_basis.py:387]: uncertain?

**Review by PeterKraus — APPROVED** [review]
Nice job on the tests, @loriab. One typo and then it can be merged?

**Comment by loriab** [comment]
general note: now that I have the correct base of `master`, azure should run and fail on mbis-3. @zachglick tracked the problem down to nondeterministic root solutions of the degenerate states. a PR will happen.

**Review by jturney — APPROVED** [review]

**Comment by loriab** [comment]
> general note: now that I have the correct base of master, azure should run and fail on mbis-3. @zachglick tracked the problem down to nondeterministic root solutions of the degenerate states. a PR will happen.

follow-up: I've removed mbis-3 from quicktests so this one can merge. note that it's a known failure in full tests for now.

**Pull request merged**